### PR TITLE
Replace eval() with getattr()/int() to prevent arbitrary code execution

### LINKED ISSUE #556: [Bug] eval() used for dynamic dispatch and data parsing — arbitrary code execution confirmed
Three uses of eval() in PyKale resolve method names, optimizer names, and annotation data from runtime strings. Arbitrary code execution was confirmed during testing — not theoretical.

Affected locations:

1. kale/pipeline/fewshot_trainer.py:114
loss, acc = eval(f"self.loss_{mode}")(feature_support, feature_query)

2. kale/pipeline/fewshot_trainer.py:149
optimizer = eval(f"torch.optim.{self.optimizer}")(self.model.parameters(), lr=self.lr)
self.optimizer is a plain string argument flowing from YACS config files.

3. kale/loaddata/video_datasets.py:79-80
if 0 <= eval(line[5]) < self.n_classes:
    data.append((line[0], eval(line[1]), eval(line[2]), eval(line[5])))
Values come from a pickled pandas DataFrame annotation file.

Confirmed exploit (location 3): running eval() on a malicious pickle value __import__('os').system('echo INJECTED_VIDEO_DATASETS') printed INJECTED_VIDEO_DATASETS to stdout with exit code 0. A real shell command executed from what should be a plain integer field.

Proof that fixes are safe — getattr() and int() produce identical results for all valid inputs:
mode=train: eval=('train_loss', 0.95) | getattr=('train_loss', 0.95) | match=True
mode=val: eval=('val_loss', 0.92) | getattr=('val_loss', 0.92) | match=True
SGD/Adam/AdamW: eval and getattr return identical optimizer types | match=True
int() results: [3, 120, 245, 7] == eval() results: [3, 120, 245, 7] | Identical: True

Suggested fixes:
Location 1: getattr(self, f"loss_{mode}")(feature_support, feature_query)
Location 2: getattr(torch.optim, self.optimizer)(self.model.parameters(), lr=self.lr)
Location 3: int(line[1]), int(line[2]), int(line[5])

Environment: PyKale 0.2.0, Python 3.11.9, Windows 11 Git Bash, PyTorch 2.6.0+cpu

### PR DESCRIPTION
**Pull request by cmanjah**
Fixes #556.

### Description
Removes three `eval()` call sites used for dynamic dispatch and annotation parsing, replacing them with `getattr()` (loss/optimizer lookup) and `int()` (frame/label parsing). Equivalent for valid inputs, but malicious values can no longer execute. Adds tests for `make_dataset`, including rejection of a code-like value.

### Status
Ready

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] In-line docstrings updated.
- [ ] Source for documentation at `docs` manually updated for new API.

### CODE REVIEW

File: kale/loaddata/video_datasets.py [hunk 1]
```diff
@@ -76,8 +76,8 @@ def make_dataset(self):
         i = 0
         input_file = pd.read_pickle(self.annotationfile_path)
         for line in input_file.values:
-            if 0 <= eval(line[5]) < self.n_classes:
-                data.append((line[0], eval(line[1]), eval(line[2]), eval(line[5])))
+            if 0 <= int(line[5]) < self.n_classes:
+                data.append((line[0], int(line[1]), int(line[2]), int(line[5])))
```
[the remark this answers is not in the document]
Comment (cmanjah) [reply kale/loaddata/video_datasets.py]: Done. `label = int(line[5])` is computed once and reused in both the range check and the append. The existing `make_dataset` tests (integer parsing + the malicious-value rejection) still pass, so behaviour is unchanged.

File: kale/pipeline/fewshot_trainer.py [hunk 2]
```diff
@@ -111,7 +111,7 @@ def compute_loss(self, feature_support, feature_query, mode="train") -> tuple:
             loss (torch.Tensor): Loss value.
             return_dict (dict): Dictionary of loss and accuracy.
         """
-        loss, acc = eval(f"self.loss_{mode}")(feature_support, feature_query)
+        loss, acc = getattr(self, f"loss_{mode}")(feature_support, feature_query)
```
[the remark this answers is not in the document]
Comment (cmanjah) [reply kale/pipeline/fewshot_trainer.py:115]: The docstring was the inaccurate part, and it's now corrected: only `loss_train` and `loss_val` exist, and `test_step` reuses the validation loss, so `mode` is `"train"` or `"val"` (there is no `loss_test`). I've reworded the docstring accordingly rather than add runtime validation — validating `mode` would change behaviour and is beyond the scope of this security fix (removing `eval`). The behaviour for an unsupported `mode` is unchanged from before this PR.

File: kale/pipeline/fewshot_trainer.py [hunk 3]
```diff
@@ -146,5 +146,5 @@ def configure_optimizers(self) -> torch.optim.Optimizer:
         """
         Configure optimizer for training. Can be modified to support different optimizers from ``torch.optim``.
         """
-        optimizer = eval(f"torch.optim.{self.optimizer}")(self.model.parameters(), lr=self.lr)
+        optimizer = getattr(torch.optim, self.optimizer)(self.model.parameters(), lr=self.lr)
```
[the remark this answers is not in the document]
Comment (cmanjah) [reply kale/pipeline/fewshot_trainer.py:150]: Agreed that a validated name with a clear `ValueError` would be friendlier. I've left it out of this PR deliberately: the previous `eval(f"torch.optim.{self.optimizer}")` also raised on an invalid name, so `getattr` preserves that behaviour, and adding validation is a functional change rather than part of removing `eval`. Worth doing as its own small enhancement (it would apply equally to the other config-driven optimizer lookups in the codebase), so I'd rather not scope-creep the security fix here.

File: kale/loaddata/video_datasets.py [hunk 4]
```diff
@@ -76,8 +76,9 @@ def make_dataset(self):
         i = 0
         input_file = pd.read_pickle(self.annotationfile_path)
         for line in input_file.values:
-            if 0 <= eval(line[5]) < self.n_classes:
-                data.append((line[0], eval(line[1]), eval(line[2]), eval(line[5])))
+            label = int(line[5])
+            if 0 <= label < self.n_classes:
+                data.append((line[0], int(line[1]), int(line[2]), label))
```
Comment (shuo-zhou) [inline kale/loaddata/video_datasets.py:81]: @xianyuanliu please confirm this is correct

Comment (xianyuanliu) [reply kale/loaddata/video_datasets.py:81]: This change looks good.

**Review by shuo-zhou — APPROVED** [review]
